src/block/Wall.php

```diff
@@ -97,13 +97,15 @@ public function onNearbyBlockChange() : void{
 	protected function recalculateConnections() : bool{
 		$changed = 0;
 
-		//TODO: implement tall/short connections - right now we only support short as per pre-1.16
-
 		foreach(Facing::HORIZONTAL as $facing){
 			$block = $this->getSide($facing);
 			if($block instanceof static || $block instanceof FenceGate || $block instanceof Thin || $block->getSupportType(Facing::opposite($facing)) === SupportType::FULL){
 				if(!isset($this->connections[$facing])){
-					$this->connections[$facing] = WallConnectionType::SHORT;
+					if($this->getSide(Facing::UP)->getTypeId() == BlockTypeIds::AIR) {
```

Comment:
```suggestion
					if($this->getSide(Facing::UP)->getTypeId() === BlockTypeIds::AIR){
```
I'm also fairly sure the logic for this is more complex than just checking for air.

Reply:
Ill do some more testing but I haven't found any issues yet.

Reply:
Yeah gonna have to add more logic, this isn't the indented state
![Minecraft 2_13_2025 11_34_25 AM](https://github.com/user-attachments/assets/10790d42-5d32-423e-a1ea-ff7d50983753)

Reply:
Defiantly issues, will continue testing and implementing more logic for vertical connections